crates/store/re_log_encoding/src/codec/wire.rs

```diff
@@ -0,0 +1,261 @@
+    fn to_bytes(&self) -> Result<Vec<u8>, CodecError> {
+        match self {
+            Self::NoData => {
+                let mut data: Vec<u8> = Vec::new();
+                MessageHeader::NO_DATA.encode(&mut data)?;
+                Ok(data)
+            }
+            Self::RecordBatch(chunk) => {
+                let mut data: Vec<u8> = Vec::new();
+                MessageHeader::RECORD_BATCH.encode(&mut data)?;
+
+                write_arrow_to_bytes(&mut data, &chunk.schema, &chunk.data)?;
+
+                Ok(data)
+            }
+        }
+    }
+
+    fn from_bytes(data: &[u8]) -> Result<Self, CodecError> {
+        let mut reader = std::io::Cursor::new(data);
+        let header = MessageHeader::decode(&mut reader)?;
+
+        match header {
+            MessageHeader::NO_DATA => Ok(Self::NoData),
+            MessageHeader::RECORD_BATCH => {
+                let (schema, data) = read_arrow_from_bytes(&mut reader)?;
+
+                let tc = TransportChunk {
+                    schema: schema.clone(),
+                    data,
+                };
+
+                Ok(Self::RecordBatch(tc))
+            }
+            _ => Err(CodecError::UnknownMessageHeader),
+        }
+    }
+}
+
+// TODO(zehiko) add support for separately encoding schema from the record batch to get rid of overhead
+// of sending schema in each transport message for the same stream of batches. This will require codec
+// to become stateful and keep track if schema was sent / received.
+/// Encode a transport chunk into a byte stream.
+pub fn encode(
+    version: re_protos::common::v0::EncoderVersion,
+    chunk: TransportChunk,
+) -> Result<Vec<u8>, CodecError> {
+    match version {
+        re_protos::common::v0::EncoderVersion::V0 => {
+            TransportMessageV0::RecordBatch(chunk).to_bytes()
+        }
+    }
+}
+
+/// Encode a `NoData` message into a byte stream. This can be used by the remote store
+/// (i.e. data producer) to signal back to the client that there's no data available.
+pub fn no_data(version: re_protos::common::v0::EncoderVersion) -> Result<Vec<u8>, CodecError> {
+    match version {
+        re_protos::common::v0::EncoderVersion::V0 => TransportMessageV0::NoData.to_bytes(),
+    }
+}
+
+/// Decode transport data from a byte stream - if there's a record batch present, return it, otherwise return `None`.
+pub fn decode(
+    version: re_protos::common::v0::EncoderVersion,
+    data: &[u8],
+) -> Result<Option<TransportChunk>, CodecError> {
+    match version {
+        re_protos::common::v0::EncoderVersion::V0 => {
+            let msg = TransportMessageV0::from_bytes(data)?;
+            match msg {
+                TransportMessageV0::RecordBatch(chunk) => Ok(Some(chunk)),
+                TransportMessageV0::NoData => Ok(None),
+            }
+        }
+    }
+}
+
+/// Helper function that serializes given arrow schema and record batch into bytes
+/// using Arrow IPC format.
+pub fn write_arrow_to_bytes<W: std::io::Write>(
```

Comment:
I guess this will be somehow re-used in a follow up PR?

Reply:
Yes, but I can remove it from this one